keywords/MobileRestClient.py

```diff
@@ -1131,24 +1130,19 @@ def add_docs(self, url, db, number, id_prefix, auth=None, channels=None, generat
             if channels is not None:
                 doc_body["channels"] = channels
 
-            data = json.dumps(doc_body)
+            if attachments_generator:
+                types.verify_is_callable(attachments_generator)
+                attachments = attachments_generator()
+                doc_body["_attachments"] = {att.name: {"data": att.data} for att in attachments}
 
             if id_prefix is None:
                 doc_id = str(uuid.uuid4())
             else:
                 doc_id = "{}_{}".format(id_prefix, i)
 
-            if auth_type == AuthType.session:
```

Comment:
Because add_docs now calls doc_obj = self.add_doc(url, db, doc_body, auth=auth). This was based on your previous comment